bosh-director/lib/bosh/director/vm_deleter.rb

```diff
@@ -17,7 +18,11 @@ def delete_for_instance(instance)
 
     def delete_vm(vm_cid)
       @logger.info('Deleting VM')
-      @error_ignorer.with_force_check { @cloud.delete_vm(vm_cid) }
+      @error_ignorer.with_force_check {
+        unless @enable_virtual_delete_vm
+          @cloud.delete_vm(vm_cid)
```

Comment:
Right so if enable_virtual_delete_vm=true then there is no way to delete VM ever. `bosh delete deployment` command will not delete VMs. I thought you actually wanted that.